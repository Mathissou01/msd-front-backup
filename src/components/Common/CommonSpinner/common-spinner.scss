@use "sass:color" as color;

.c-CommonSpinner {
  position: relative;
  display: inline-block;
  width: 80px;
  height: 80px;
  transform: scale(75%);

  div {
    position: absolute;
    display: block;
    box-sizing: border-box;
    width: 64px;
    height: 64px;
    margin: 8px;
    border: 8px solid;
    border-color: $expert-blue transparent transparent;
    border-radius: 50%;
    animation: common-spinner 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  }

  div:nth-child(1) {
    animation-delay: -0.45s;
  }

  div:nth-child(2) {
    animation-delay: -0.3s;
  }

  div:nth-child(3) {
    animation-delay: -0.15s;
  }

  @keyframes common-spinner {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }

  &__Wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;

    &_cover {
      position: absolute;
      width: 100%;
      height: 100%;
      pointer-events: none;
    }
  }
}
