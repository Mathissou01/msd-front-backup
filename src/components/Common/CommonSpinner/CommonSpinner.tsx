import React from "react";
import "./common-spinner.scss";

interface ICommonSpinnerProps {
  isCover?: boolean;
}

export default function CommonSpinner({
  isCover = false,
}: ICommonSpinnerProps) {
  return (
    <div
      className={`c-CommonSpinner__Wrapper ${
        isCover ? "c-CommonSpinner__Wrapper_cover" : ""
      }`}
    >
      <div className="c-CommonSpinner" data-testid="common-spinner">
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  );
}
