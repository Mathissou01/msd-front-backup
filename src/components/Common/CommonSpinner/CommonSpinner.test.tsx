import { render } from "@testing-library/react";
import CommonSpinner from "./CommonSpinner";

it("renders", () => {
  const { container } = render(<CommonSpinner />);
  expect(container).toMatchSnapshot();
});

it("renders in cover mode", () => {
  const { container } = render(<CommonSpinner isCover={true} />);
  expect(container).toMatchSnapshot();
});
