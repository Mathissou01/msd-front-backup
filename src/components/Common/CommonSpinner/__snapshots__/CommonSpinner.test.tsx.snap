// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    class="c-CommonSpinner__Wrapper "
  >
    <div
      class="c-CommonSpinner"
      data-testid="common-spinner"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`renders in cover mode 1`] = `
<div>
  <div
    class="c-CommonSpinner__Wrapper c-CommonSpinner__Wrapper_cover"
  >
    <div
      class="c-CommonSpinner"
      data-testid="common-spinner"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
